itest/rust/src/builtin_tests/containers/dictionary_test.rs

```diff
@@ -383,6 +383,35 @@ fn dictionary_iter() {
     assert_eq!(map, map2);
 }
 
+#[itest]
+fn dictionary_iter_size_hint() {
+    // Test a completely empty dict
+    let dictionary = Dictionary::new();
+    let iter = dictionary.iter_shared();
+    assert_eq!(iter.size_hint(), (0, Some(0)));
+    // ...then one full being emptied
+    let dictionary = dict! {
+        "foo": 0,
+        "bar": true,
+        "baz": "foobar",
+        "nil": Variant::nil(),
+    };
+    let mut dictionary_clone = dictionary.clone();
+    let mut iter = dictionary.iter_shared();
+    assert_eq!(iter.size_hint(), (4, Some(4)));
+    iter.next();
+    iter.next();
+    iter.next();
+    assert_eq!(iter.size_hint(), (1, Some(1)));
+    iter.next();
+    assert_eq!(iter.size_hint(), (0, Some(0)));
+    iter.next();
+    assert_eq!(iter.size_hint(), (0, Some(0)));
+    // Now make sure that size hint might change here
+    dictionary_clone.insert("new_key", "soma_val");
+    assert_eq!(iter.size_hint(), (1, Some(1)));
+}
```

Comment:
Also here, some empty lines would really help readability.
Comments should ideally end with fullstop (rarely colon or ellipsis).
```suggestion
#[itest]
fn dictionary_iter_size_hint() {
    // Test a completely empty dict.
    let dictionary = Dictionary::new();
    let iter = dictionary.iter_shared();
    assert_eq!(iter.size_hint(), (0, Some(0)));
    
    // Test a full dictionary being emptied.
    let dictionary = dict! {
        "foo": 0,
        "bar": true,
        "baz": "foobar",
        "nil": Variant::nil(),
    };
    
    let mut dictionary_clone = dictionary.clone();
    let mut iter = dictionary.iter_shared();
    assert_eq!(iter.size_hint(), (4, Some(4)));
    
    iter.next();
    iter.next();
    iter.next();
    assert_eq!(iter.size_hint(), (1, Some(1)));
    
    iter.next();
    assert_eq!(iter.size_hint(), (0, Some(0)));
    
    iter.next();
    assert_eq!(iter.size_hint(), (0, Some(0)));
    
    // Insertion while iteration is allowed and might change size hint.
    dictionary_clone.insert("new_key", "soma_val");
    assert_eq!(iter.size_hint(), (1, Some(1)));
}
```